lib/sdam/topology.js

```diff
@@ -310,14 +310,29 @@ class Topology extends EventEmitter {
       options
     );
 
+    const isSharded = this.description.type === TopologyType.Sharded;
+    const session = options.session;
+    const transaction = session && session.transaction;
+
+    if (isSharded && transaction && transaction.server) {
```

Comment:
`mongos.js` has a protection here to ensure that the pinned server is actually connected. Should we do that here?

Reply:
I think that's because the old mongos doesn't have proper server selection. Anyway, I think it's too defensive to check here and I'd rather fix this holistically in our new topology if the issue comes up.